Homework13/Tests/WebTableTests.cs

```diff
@@ -0,0 +1,138 @@
+                Assert.AreEqual(filteredResultRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+                Assert.AreEqual(filteredRowsCountWithSearchedValue, resultRowsCountWithSearchedValue);
+            });
+        }
+
+        // Test3: Add new entry.
+        [Test]
+        public void AddNewEntry()
+        {
+            GenericPages.WebTablePage.ClickAddNewEntryButton();
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameInput(firstName);
+            GenericPages.WebTablePage.FillInLastNameInput(lastName);
+            GenericPages.WebTablePage.FillInEmailInput(email);
+            GenericPages.WebTablePage.FillInAgeInput(age);
+            GenericPages.WebTablePage.FillInSalaryInput(salary);
+            GenericPages.WebTablePage.FillInDepartmentInput(department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of new entry and compare it with expected.
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfCollectionElements(GenericPages.WebTablePage.GetNewRow()));
+        }
+
+        // Test4: Edit first entry.
+        [Test]
+        public void EditFirstEntry()
+        {
+            GenericPages.WebTablePage.ClickEditEntryButton();
+
+            // Clear form inputs.
+            GenericPages.WebTablePage.ClearFirstNameInput();
+            GenericPages.WebTablePage.ClearLastNameInput();
+            GenericPages.WebTablePage.ClearEmailInput();
+            GenericPages.WebTablePage.ClearAgeInput();
+            GenericPages.WebTablePage.ClearSalaryInput();
+            GenericPages.WebTablePage.ClearDepartmentInput();
+
+            // Fill in form inputs.
+            GenericPages.WebTablePage.FillInFirstNameInput(firstName);
+            GenericPages.WebTablePage.FillInLastNameInput(lastName);
+            GenericPages.WebTablePage.FillInEmailInput(email);
+            GenericPages.WebTablePage.FillInAgeInput(age);
+            GenericPages.WebTablePage.FillInSalaryInput(salary);
+            GenericPages.WebTablePage.FillInDepartmentInput(department);
+            GenericPages.WebTablePage.ClickSubmitButton();
+
+            // Get text of updated entry and compare it with expected.
+            Assert.AreEqual(entryValue, GenericPages.WebTablePage.GetTextOfCollectionElements(GenericPages.WebTablePage.GetEditedRow()));
+        }
+
+        // Test5: Delete first entry.
+        [Test]
+        public void DeleteFirstEntry()
+        {
+            // Get the first name from the first row.
+            var firstName1 = GenericPages.WebTablePage.GetFirstName1Value();
+
+            // Get the rows number with first name from condition before the delete opetarion.
+            var initialRows = GenericPages.WebTablePage.GetRows();
+            var rowsWithInitialValue = GenericPages.WebTablePage.GetCountOfFilteredRows(GenericPages.WebTablePage.GetListOfNotEmptyRows(initialRows), firstName1);
```

Comment:
GetCountOfFilteredRows is not appropriate as there are no filtering actions before -